homeassistant/components/light/homematic.py

```diff
@@ -0,0 +1,112 @@
+"""
+The homematic light platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.homematic/
+
+Important: For this platform to work the homematic component has to be
+properly configured.
+
+Configuration:
+
+light:
+  - platform: homematic
+    addresss: <Homematic addresss for device> # e.g. "JEQ0XXXXXXX"
+    name: <User defined name> (optional)
+    button: n (integer of channel to map, device-dependent)
+"""
+
+import logging
+from homeassistant.components.light import (ATTR_BRIGHTNESS, Light)
+from homeassistant.const import STATE_UNKNOWN
+import homeassistant.components.homematic as homematic
+
+_LOGGER = logging.getLogger(__name__)
+
+# List of component names (string) your component depends upon.
+DEPENDENCIES = ['homematic']
+
+
+def setup_platform(hass, config, add_callback_devices, discovery_info=None):
+    """Setup the platform."""
+    return homematic.setup_hmdevice_entity_helper(HMLight,
```

Comment:
I think that this helper is not needed. Instead you should limit each platform to only be setup by the passed in discovery info. That way you know that the incoming info is good and also that `HOMEMATIC` object is set up (because that is what triggered discovery).

```python
def setup_platform(hass, config, add_devices, discovery_info=None):
    """"Setup the platform."""
    if discovery_info is None:
        return
    add_devices([HMLight(discovery_info)])
```